effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -49,8 +49,8 @@ object Namer extends Phase[Parsed, NameResolved] {
    * Produces a cyclic import error when this is already the case
    */
   private def recursiveProtect[R](mod: ModuleDecl)(body: => R)(using Context): R = {
-    if (currentlyNaming.value.contains(mod)) {
-      val cycle = mod :: currentlyNaming.value.takeWhile(_ != mod).reverse
+    if (currentlyNaming.value.exists { m => m.path == mod.path }) {
+      val cycle = mod :: currentlyNaming.value.takeWhile{ m => m.path != mod.path }.reverse
```

Comment:
```suggestion
    if (currentlyNaming.value.exists { m => m.span.source == mod.span.source }) {
      val cycle = mod :: currentlyNaming.value.takeWhile{ m => m.span.source != mod.span.source }.reverse
```

or even just store the source (and not the whole `ModuleDecl`)